File: week_2/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int get_letters(string text);
int get_words(string text);
int get_sentences(string text);


//main function
int main()
{
    string text = get_string("Text: ");

    //pass arguments to helper functions
    int letters = get_letters(text);
    int words = get_words(text);
    int sentences = get_sentences(text);

    // Calculate the letters and sentences per 100 words of the text with floats once helpers are done 100 * the variable / words
    float L = (float)letters / words * 100;
    float S = (float)sentences / words * 100;

    //Coleman-Liau index
    float index = 0.0588 * L - 0.296 * S - 15.8;
    int final_grade = round(index);

    if (final_grade < 1)
    {
        //Don't add the variable when calling Before Grade 1, otherwise, the number may be well below 1
        printf("Before Grade 1\n");
    }
    else if (final_grade >= 16)
    {
        //Don't add the variable when calling 16+, otherwise, you will return the exact number which in this case, you don't want to return
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", final_grade);
    }
}

//add helper functions
// count letters
int get_letters(string text)
{
    int count = 0;

    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] >= 'A' && text[i] <= 'z')
        {
            count++;
        }
    }
    return count;
}

// //count words
int get_words(string text)
{
    //the first word will always be skipped in the count, so start at 1
    int count = 1;

    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == ' ')
        {
            count++;
        }
    }
    return count;
}
// //count sentences
int get_sentences(string text)
{
    int count = 0;

    for (int i = 0; i <= strlen(text); i++)
    {
        if (text[i] == '.' || text[i] == '?' || text[i] == '!')
        {
            count++;
        }
    }
    return count;
}

File: pset6/readability/readability.py
import math

# Create helpers


def get_letters(text):
    # Count letters
    letter_count = 0
    for l in text.lower():
        if l.isalnum() == True:
            letter_count += 1
    return letter_count


def get_words(text):
    # Count words
    word_count = 1
    for i in range(len(text)):
        if(text[i] == ' ' or text[i] == '\n' or text[i] == '\t'):
            word_count += 1
    return word_count


def get_sentences(text):
    # Count sentences
    sentence_count = 0
    for i in text:
        if i == '.' or i == '!' or i == '?':
            sentence_count += 1
    return sentence_count


def main():
    while True:
        # Ask for user input
        user_input = input("Enter text: ")

        # Call helper functions and pass same text to each
        letters = get_letters(user_input)
        words = get_words(user_input)
        sentences = get_sentences(user_input)

        # Calculate the letters and sentences per 100 words of the text with floats once helpers are done 100 * the variable / words
        L = letters / words * 100
        S = sentences / words * 100

        # Coleman-Liau index
        index = 0.0588 * L - 0.296 * S - 15.8
        floating_index = float(index)
        print(floating_index)

        # Round floating_index for final grade
        final_grade = round(floating_index)

        if final_grade < 1:
            return(print('Before Grade 1'))
        elif final_grade >= 16:
            return(print('Grade 16+'))
        else:
            return(print(f'Grade {final_grade}'))


# Call main function
main()
